test/src/org/labkey/test/components/targetedms/QCPlotsWebPart.java

```diff
@@ -748,6 +747,9 @@ public String getLabel()
         TPAREARATIO("Transition/Precursor Area Ratio", true),
         TPAREAS("Transition & Precursor Areas", true),
         MASSACCURACY("Mass Accuracy", true),
+        IRTINTERCEPT("iRT Intercept", false),
+        IRTSLOPE("iRT Slope", false),
+        IRTCORRELATION("iRT Correlation", false),
```

Comment:
The "hasData" flag seems like a bad thing to put into what's supposed to be a reusable component. The iRT metrics have data in some (but certainly not all) sample files. It seems like the test, which controls which file(s) have been loaded should be responsible for knowing what metrics should have data.

Reply:
how about i add one more parameter isOptionalMetric and use that in test? or u think tests should handle that?

Reply:
It looks like the only place that uses hasData() right now is TargetedMSQCTest.testQCPlotInputs(). It calls it four times but it'll never enter the if() block unless the first call to hasData() return true.

I'd recommend making that part of the test check whatever set of metrics it wants to be there and removing the hasData setting altogether.